Foundation/FileHandle.swift

```diff
@@ -109,7 +109,7 @@ open class FileHandle : NSObject, NSSecureCoding {
         
```

Comment:
The earlier code paths ensure that the total will never be negative. Regardless, I updated this to cover all cases and added an assertion in the unreachable case.